### PR TITLE
Base module for APM Tracing

### PR DESCRIPTION
**Pull request by tlrx**

### CODE REVIEW

**Review by idegtiarenko — APPROVED** [review]

File: server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java [hunk 1]
```diff
@@ -142,7 +145,8 @@ public TransportSearchAction(
         ActionFilters actionFilters,
         IndexNameExpressionResolver indexNameExpressionResolver,
         NamedWriteableRegistry namedWriteableRegistry,
-        ExecutorSelector executorSelector
+        ExecutorSelector executorSelector,
+        TracingPlugin.Tracer tracer
```
Comment (dimitris-athanasiou) [inline server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:149]: Should we add this here? If we start by tracing tasks, we won't need this dependency in the search action.

Comment (DaveCTurner) [reply server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:149]: It at least demonstrates that we're creating and registering it when creating the `Node`. We can drop it when we use the injected thing elsewhere.

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java (hunk 1 above)
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java:149]: > It at least demonstrates that we're creating and registering it

Yes, the purpose was to make the `Tracer` instance available for the next steps, whatever/wherever they will be

**Review by dimitris-athanasiou — APPROVED** [review]
LGTM2

**Pull request merged**